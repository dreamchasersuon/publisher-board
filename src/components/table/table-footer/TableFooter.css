.TableFooter-button {
  width: 100%;
  padding: 15px;
  margin: 15px 0;
  border-radius: 4px;
  font-size: 14px;
  border: 1px solid #e5e5e5;
  background-color: transparent;
  cursor: pointer;
  outline: none;
  transition: all 0.6s cubic-bezier(0.39, 0.575, 0.565, 1);
}

.TableFooter-button:hover {
  background-color: #fafafa;
}
