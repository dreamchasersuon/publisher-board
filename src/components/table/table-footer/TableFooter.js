import React from 'react';
import PropTypes from 'prop-types';
import Spinner from '../../spinner/Spinner';
import './TableFooter.css';

export default function TableFooter({
  isReachingEnd,
  isLoadingMore,
  loadMore,
}) {
  return (
    <tfoot>
      <tr>
        <td colSpan="100%">
          <button onClick={loadMore} className="TableFooter-button">
            {isLoadingMore ? <Spinner /> : 'LOAD MORE'}
          </button>
        </td>
      </tr>
    </tfoot>
  );
}

TableFooter.propTypes = {
  isReachingEnd: PropTypes.bool,
  isLoadingMore: PropTypes.bool,
  loadMore: PropTypes.func,
};
