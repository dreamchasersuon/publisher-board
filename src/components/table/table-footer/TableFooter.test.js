import React from 'react';
import snapshotDiff from 'snapshot-diff';
import TableFooter from './TableFooter';

const loadMore = jest.fn();

test('renders table footer with spinner or disabled', () => {
  expect(
    snapshotDiff(
      <TableFooter isLoadingMore loadMore={loadMore} />,
      <TableFooter loadMore={loadMore} isReachingEnd />,
    ),
  ).toMatchSnapshot();
});
