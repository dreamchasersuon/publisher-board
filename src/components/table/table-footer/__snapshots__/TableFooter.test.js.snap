// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders table footer with spinner or disabled 1`] = `
"Snapshot Diff:
- <TableFooter isLoadingMore={true} loadMore={[Function mockConstructor]} />
+ <TableFooter isReachingEnd={true} loadMore={[Function mockConstructor]} />

@@ -3,14 +3,13 @@
      <td
        colSpan=\\"100%\\"
      >
        <button
          className=\\"TableFooter-button\\"
+         disabled={true}
          onClick={[Function mockConstructor]}
        >
-         <i
-           className=\\"fas fa-spinner fa-spin\\"
-         />
+         LOAD MORE
        </button>
      </td>
    </tr>
  </tfoot>"
`;
